### ISSUE TITLE
Implement Cat 4 validation error handling and record rejection

### ISSUE THREAD

**Issue by andrew-jameson** [raft-tech/TANF-app#3177]
## Background ##
During review of [PR #3134](https://github.com/raft-tech/TANF-app/pull/3134#issuecomment-2338792743), it was identified that the **Category 4 (cat4) validation logic** in `case_consistency_validator.py` does not currently **increment `num_errors`** or reject records when cat4 validation fails.

Per follow-up with the OFA team (Dec. 9), the system should **reject any case records** with cat4 validation issues. This ticket corrects that behavior and ensures database seeding creates consistent test data while still allowing for controlled cat4 failure scenarios.

### Why It Matters ##
- Ensures proper enforcement of **cat4 validation rules**
- Prevents invalid records from being accepted into the system
- Aligns behavior with OFA expectations and validation logic
- Supports consistent testing of both valid and failing cat4 cases

### What's Changing ##
- Adds the missing `num_errors` increment within the `validate()` function of `case_consistency_validator.py`
- Updates the database seeding utility to generate **valid** cat4 test cases while allowing controlled failure injection for testing

### Acceptance Criteria ###
- [ ] `num_errors` is incremented when cat4 validation fails, causing rejection of invalid records
- [ ] `db_seed.py` generates cases that **pass cat4 validation**
- [ ] Ability to optionally generate **cat4 failures** for testing (stretch goal)
- [ ] Testing Checklist is completed and all tests pass
- [ ] README or validator documentation is updated if needed
- [ ] [Release digest](https://app.gitbook.com/o/Yd4Wv0Fi89kSKpmakWrn/s/JSIbkxujKUxhKv2cXFEg/) has been updated with testing steps & relevant media  

### Usability Testing Criteria ###
N/A

### Notes ##
- Per discussion with Alex (OFA), all records with cat4 issues **must be rejected** once this is in place
- Future testing should cover both passing and failing cat4 cases to confirm expected behavior

### Supporting Documentation ##
- [PR #3134 Validation Issue Comment](https://github.com/raft-tech/TANF-app/pull/3134#issuecomment-2338792743)
- `case_consistency_validator.py` reference line 160

**Comment by lhuxraft** [comment]
Discuss if behavior is thought about in the right way in office hours
